applications/StructuralMechanicsApplication/custom_elements/truss_element_3D2N.cpp

```diff
@@ -303,8 +303,23 @@ void TrussElement3D2N::Calculate(const Variable<Matrix>& rVariable, Matrix& rOut
         }
         noalias(rOutput) = transformation_matrix;
     }
+    if (rVariable == MEMBRANE_PRESTRESS) {
+        std::vector< Vector > prestress_matrix;
+        CalculateOnIntegrationPoints(PK2_STRESS_VECTOR,prestress_matrix,rCurrentProcessInfo);

```

Comment:
Does that mean that when you have orthotropic pre-stress that you just define the direction from outside? Is there a common script to do that?

Reply:
no, a truss element can only have a 1-dimensional pre-stress

Reply:
and how is this done in the membrane, shell or solid formulations.

Reply:
in these cases, you have to use the projection utility, if you don't have isotropic pre-stress, to define the pre-stress axes. And the element handles the transformation of bases

Reply:
ok nice!